api/server/router/system/backend.go

```diff
@@ -10,12 +10,24 @@ import (
 	"github.com/docker/docker/api/types/swarm"
 )
 
+// DiskUsageOptions holds parameters for system disk usage query.
+type DiskUsageOptions struct {
+	// Containers controls whether container disk usage should be computed.
+	Containers bool
+
+	// Images controls whether image disk usage should be computed.
+	Images bool
+
+	// Volumes controls whether volume disk usage should be computed.
+	Volumes bool
```

Comment:
So, I was looking if an alternative could be  to use a `map[DiskUsageObject]bool` to indicate which objects to collect, then iterate over that list. 

I did some quick dabbling with that idea; let me push that to a branch, and we can chat Tomorrow